examples/pytorch/tensor_parallel/train.py

```diff
@@ -0,0 +1,75 @@
+import lightning as L
+import torch
+import torch.nn.functional as F
+from data import RandomTokenDataset
+from lightning.pytorch.strategies import ModelParallelStrategy
+from model import ModelArgs, Transformer
+from parallelism import parallelize
+from torch.distributed.tensor.parallel import loss_parallel
+from torch.utils.data import DataLoader
+
+
+class Llama2(L.LightningModule):
+    def __init__(self):
+        super().__init__()
+        self.model_args = ModelArgs(vocab_size=32000)
+        self.model = Transformer(self.model_args)
+
+    def configure_model(self):
+        # User-defined function that applies the desired parallelizations specific to the model
+        # (TP, FSDP2, activation checkpointing, ...)
+        parallelize(self.model, device_mesh=self.device_mesh)
+
+    def on_train_start(self) -> None:
+        self.model.init_weights()
+
+    def training_step(self, batch):
+        inputs = batch[:, :-1]
+        labels = batch[:, 1:]
+        output = self.model(inputs)
+        with loss_parallel():
```

Comment:
TIL. Are you also enabling this for backward? https://github.com/pytorch/pytorch/blob/5fb11cda4fe60c1a7b30e6c844f84ce8933ef953/torch/distributed/tensor/parallel/loss.py#L35